File: regexlib/javascript_test_file/regexlib_7940.js
/* 7940
 * Last.*?(\d+.?\d*)
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"Last"+"1"*1024+"! _1SLQ_2"
 */
var REGEX = /Last.*?(\d+.?\d*)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'Last' + '1'.repeat(i*1) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re_test_file/regexlib_7940.py
# 7940
# Last.*?(\d+.?\d*)
# EXPONENT
# nums:4
# EXPONENT AttackString:"Last"+"1"*512+"! _1SLQ_2"

import re
from time import perf_counter

regex = """Last.*?(\d+.?\d*)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "Last" + "1" * i * 1 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")